maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/mapper/IncomeEvidenceMapper.java

```diff
@@ -0,0 +1,241 @@
+package uk.gov.justice.laa.crime.orchestration.mapper;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.justice.laa.crime.common.model.evidence.*;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.FinancialAssessmentIncomeEvidence;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.MaatApiAssessmentResponse;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.MaatApiUpdateAssessment;
+import uk.gov.justice.laa.crime.enums.EmploymentStatus;
+import uk.gov.justice.laa.crime.enums.MagCourtOutcome;
+import uk.gov.justice.laa.crime.exception.ValidationException;
+import uk.gov.justice.laa.crime.orchestration.dto.WorkflowRequest;
+import uk.gov.justice.laa.crime.orchestration.dto.maat.*;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.FinAssIncomeEvidenceDTO;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.RepOrderDTO;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.FinancialAssessmentDTO;
+import uk.gov.justice.laa.crime.util.DateUtil;
+import uk.gov.justice.laa.crime.util.NumberUtils;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static java.util.Optional.ofNullable;
+
+@Component
+@RequiredArgsConstructor
+public class IncomeEvidenceMapper {
+
+    private final UserMapper userMapper;
+    private final MeansAssessmentMapper meansAssessmentMapper;
+
+    public ApiCreateIncomeEvidenceRequest workflowRequestToApiCreateIncomeEvidenceRequest(WorkflowRequest workflowRequest) {
+        ApplicationDTO application = workflowRequest.getApplicationDTO();
+        ApiApplicantDetails partnerDetails = getPartnerDetails(application.getApplicantLinks());
+        Collection<AssessmentSectionSummaryDTO> initSectionSummaries = application.getAssessmentDTO()
+                .getFinancialAssessmentDTO().getInitial().getSectionSummaries();
+
+        return new ApiCreateIncomeEvidenceRequest()
+                .withMagCourtOutcome(MagCourtOutcome.getFrom(application.getMagsOutcomeDTO().getOutcome()))
+                .withApplicantDetails(getApplicantDetails(application.getApplicantDTO()))
+                .withPartnerDetails(partnerDetails)
+                .withApplicantPensionAmount(getPensionAmount(initSectionSummaries, false))
```

Comment:
Yeh that's a good point, I think I wrote it like that as I thought it would make getPensionDetails() more useful in the future as it would encompass getting the pension details and returning the amount. It does make it less efficient and we can always refactor in the future if needed. So I've extracted the pension details filter out of getPensionAmount().